conans/client/graph/graph_builder.py

```diff
@@ -99,22 +103,22 @@ def _expand_require(self, require, node, graph, profile_host, profile_build, gra
 
     @staticmethod
     def _conflicting_version(require, node,
-                             prev_require, prev_node, prev_ref, base_previous):
+                             prev_require, prev_node, prev_ref, base_previous, resolve_prereleases):
         version_range = require.version_range
         prev_version_range = prev_require.version_range if prev_node is None else None
         if version_range:
             # TODO: Check user/channel conflicts first
             if prev_version_range is not None:
                 pass  # Do nothing, evaluate current as it were a fixed one
             else:
-                if prev_ref.version in version_range:
+                if version_range.contains(prev_ref.version, resolve_prereleases):
                     require.ref = prev_ref
                 else:
                     raise GraphError.conflict(node, require, prev_node, prev_require, base_previous)
 
         elif prev_version_range is not None:
-            # TODO: CHeck user/channel conflicts first
-            if require.ref.version not in prev_version_range:
+            # TODO: Check user/channel conflicts first
+            if not prev_version_range.contains(require.ref.version):
```

Comment:
why this doesn't crash if not provided the arg? the intention is that it is mandatory to always pass the "resolve_prereleases" arg.